src/frontend/github-oauth.ts

```diff
@@ -148,14 +151,39 @@ export default (opts: OAuthOptions): GithubOAuth => {
 		checkGithubAuth: async (req: Request, res: Response, next: NextFunction) => {
 			const traceLogsEnabled = await booleanFlag(BooleanFlags.TRACE_LOGGING, false);
 			const tracer = new Tracer(logger.child(opts), "checkGithubAuth", traceLogsEnabled);
+			try {
+				const { githubToken } = req.session;
+				if (!githubToken) {
+					tracer.trace("github token missing, calling login()");
+					throw "Missing github token";
+				}
+				tracer.trace("found github token in session. validating token with API.");
 
-			if (!req.session.githubToken) {
-				tracer.trace("found github token in session, calling login()");
-				res.locals.redirect = req.originalUrl;
-				return login(req, res);
+				await axios.get(`https://api.${envVars.GITHUB_DOMAIN}`, {
+					headers: {
+						Authorization: `Bearer ${githubToken}`
+					}
+				});
+
+				req.log.debug(`Github token is valid, continuing...`);
+
+				// Everything's good, set it to res.locals
+				res.locals.githubToken = githubToken;
+				// TODO: Not a great place to put this, but it'll do for now
+				res.locals.github = GithubApi({ auth: githubToken });
+				return next();
+			} catch (e) {
+				req.log.debug(`Github token is not valid.`);
+				// If its a GET request, we can redirect to login and try again
+				if (req.method == "GET") {
```

Comment:
We can only redirect to the github login flow if it's a GET request, for any other requests, we need to fail the request and handle that failure on the front-end side.  This is not implemented here as this was always an issue, but we should have a better way of handling API failures in the UI.